src/insert.rs

```diff
@@ -84,8 +84,8 @@ impl InsertState {
     }
 
     fn terminated(&mut self) {
-        debug_assert!(matches!(self, InsertState::Active { .. }));
```

Comment:
Shall we keep this assertion for either NotStarted or Active?

Reply:
No need, `unreachable!()` is an assertion which doesn't go away in debug build